service/workspaces: acctests/sweepers return error when resource not found

### Affected AccTests:

* TestAccAwsWorkspacesWorkspace_basic
* TestAccAwsWorkspacesWorkspace_tags
### Affected Sweepers

<!--- Please list the affected resources and data sources. --->

* aws_workspaces_workspace
* aws_workspaces_directory

### Expected Behavior

The acctests/sweepers should not return an error, since we are waiting for the absence of the resource.

### Actual Behavior

The acctest/sweeper returns the error `workspace "ws-g7mcyy6c6" was not terminated: couldn't find resource (21 retries)`.

### How to Fix

When waiting for a resource to be removed, `WaitForState()` should have an empty `Target`, and the state function should return `nil, "", nil`

FYI this effects the testing as well:

```
=== CONT  TestAccAwsWorkspacesWorkspace_basic
TestAccAwsWorkspacesWorkspace_basic: resource_aws_workspaces_workspace_test.go:63: Step 1/2 error: terraform failed: exit status 1
stderr:
Error: workspace "ws-g7mcyy6c6" is not available: timeout while waiting for state to become 'AVAILABLE' (last state: 'PENDING', timeout: 30m0s)
TestAccAwsWorkspacesWorkspace_basic: testing_new.go:22: WARNING: destroy failed, so remote objects may still exist and be subject to billing
TestAccAwsWorkspacesWorkspace_basic: testing_new.go:22: failed to destroy: terraform failed: exit status 1
stderr:
Error: error deleting WorkSpaces Directory (d-9267063c3e): error deregistering WorkSpaces Directory "d-9267063c3e": OperationNotSupportedException: There are WorkSpaces assigned to this directory. You must remove the WorkSpaces before you deregister the directory.
--- FAIL: TestAccAwsWorkspacesWorkspace_basic (2095.57s)

=== CONT  TestAccAwsWorkspacesWorkspace_tags
TestAccAwsWorkspacesWorkspace_tags: testing_new.go:22: WARNING: destroy failed, so remote objects may still exist and be subject to billing
TestAccAwsWorkspacesWorkspace_tags: testing_new.go:22: failed to destroy: terraform failed: exit status 1
stderr:
Error: workspace "ws-bzvj5njdh" was not terminated: couldn't find resource (21 retries)
--- FAIL: TestAccAwsWorkspacesWorkspace_tags (1672.55s)
```

--- a/aws/internal/service/workspaces/waiter/status.go
+++ b/aws/internal/service/workspaces/waiter/status.go
@@ -34,10 +34,17 @@ func WorkspaceState(conn *workspaces.WorkSpaces, workspaceID string) resource.St
 		}
 
 		if len(output.Workspaces) == 0 {
-			return nil, workspaces.WorkspaceStateTerminated, nil
+			return nil, "", nil
 		}
 
 		workspace := output.Workspaces[0]
+
+		// https://docs.aws.amazon.com/workspaces/latest/api/API_TerminateWorkspaces.html
+		// State TERMINATED is overridden with TERMINATING to catch up directory metadata clean up.
+		if aws.StringValue(workspace.State) == workspaces.WorkspaceStateTerminated {
+			return workspace, workspaces.WorkspaceStateTerminating, nil
+		}
+
 		return workspace, aws.StringValue(workspace.State), nil
 	}
 }

--- a/aws/internal/service/workspaces/waiter/waiter.go
+++ b/aws/internal/service/workspaces/waiter/waiter.go
@@ -109,9 +109,7 @@ func WorkspaceTerminated(conn *workspaces.WorkSpaces, workspaceID string) (*work
 			workspaces.WorkspaceStateTerminating,
 			workspaces.WorkspaceStateError,
 		},
-		Target: []string{
-			workspaces.WorkspaceStateTerminated,
-		},
+		Target:  []string{},
 		Refresh: WorkspaceState(conn, workspaceID),
 		Timeout: WorkspaceTerminatedTimeout,
 	}

--- a/aws/resource_aws_workspaces_workspace_test.go
+++ b/aws/resource_aws_workspaces_workspace_test.go
@@ -284,6 +284,39 @@ func TestAccAwsWorkspacesWorkspace_validateUserVolumeSize(t *testing.T) {
 	})
 }
 
+func TestAccAwsWorkspacesWorkspace_recreate(t *testing.T) {
+	var v workspaces.Workspace
+	rName := acctest.RandString(8)
+
+	resourceName := "aws_workspaces_workspace.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccPreCheckWorkspacesDirectory(t)
+			testAccPreCheckAWSDirectoryServiceSimpleDirectory(t)
+			testAccPreCheckHasIAMRole(t, "workspaces_DefaultRole")
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsWorkspacesWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWorkspacesWorkspaceConfig(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAwsWorkspacesWorkspaceExists(resourceName, &v),
+				),
+			},
+			{
+				Taint:  []string{resourceName}, // Force workspace re-creation
+				Config: testAccWorkspacesWorkspaceConfig(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAwsWorkspacesWorkspaceExists(resourceName, &v),
+				),
+			},
+		},
+	})
+}
+
 func testAccCheckAwsWorkspacesWorkspaceDestroy(s *terraform.State) error {
 	conn := testAccProvider.Meta().(*AWSClient).workspacesconn
 
@@ -300,7 +333,7 @@ func testAccCheckAwsWorkspacesWorkspaceDestroy(s *terraform.State) error {
 		}
 
 		if len(resp.Workspaces) == 0 {
-			return fmt.Errorf("workspace %q was not terminated", rs.Primary.ID)
+			return nil
 		}
 		ws := resp.Workspaces[0]
 

